### PR TITLE
Fix building gtest on FreeBSD/NetBSD:

### PR DESCRIPTION
**Pull request by bsdkurt**
Tested on OpenBSD/amd64 and FreeBSD/aarch64 with #50

### CODE REVIEW

File: test/fmw/gtest/include/gtest/internal/gtest-port.h [hunk 1]
```diff
@@ -263,8 +263,8 @@
 # define GTEST_OS_HPUX 1
 #elif defined __native_client__
 # define GTEST_OS_NACL 1
-#elif defined __OpenBSD__
-# define GTEST_OS_OPENBSD 1
+#elif defined(__OpenBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__)
```
Comment (battleblow) [inline test/fmw/gtest/include/gtest/internal/gtest-port.h:266]: Looks like one of these OpenBSDs should read NetBSD

**Comment by krytarowski** [comment]
In general gtest has upstream and upstreamed NetBSD support cleanly. For some reason projects inline snapshot of them into 3rd party code.

File: test/fmw/gtest/include/gtest/internal/gtest-port.h (hunk 1 above)
Comment (battleblow) [reply test/fmw/gtest/include/gtest/internal/gtest-port.h:266]: I pushed a change to fix this.

**Comment by battleblow** [comment]
> In general gtest has upstream and upstreamed NetBSD support cleanly. For some reason projects inline snapshot of them into 3rd party code.

The OpenJDK code base does have bundled versions of many libraries.  The configuration script allows you to use the system version instead in most cases, but not all.  This is one of the cases where that isn't true.

I think a discussion of bundling vs external dependencies as a general practice is beyond the scope of this code change :)

**Comment by krytarowski** [comment]
OK, my suggestion expressed now more literally - please check upstream gtest, cherry picking patches, instead of reworking the code differently downstream.

File: test/fmw/gtest/include/gtest/internal/gtest-port.h (hunk 1 above)
Comment (bsdkurt) [reply test/fmw/gtest/include/gtest/internal/gtest-port.h:266]: Thanks Greg.

**Comment by battleblow** [comment]
FWIW, this will happen naturally.  Already JDK 13 contains a newer version of gtest that doesn't need any changes to support BSD.  I think for now I'd rather wait for this sort of thing to happen.  Pulling patches from upstream can be a good solution, but I fear we'd potentially be spending a lot of time trying to figure out a patch that both adds what we want and is compatible with this older version of gtest.  This will also mean more merge conflicts if this newer version does get merged back.  So there are some trade offs to consider here.

**Pull request merged**